fenixedu-ist-integration/src/main/java/pt/ist/fenixedu/integration/domain/teacher/summaries/FenixEduISTSummariesContextListener.java

```diff
@@ -0,0 +1,44 @@
+package pt.ist.fenixedu.integration.domain.teacher.summaries;
+
+import java.util.Calendar;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+
+import org.fenixedu.academic.domain.OccupationPeriodType;
+import org.fenixedu.academic.domain.Summary;
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.bennu.core.signals.DomainObjectEvent;
+import org.fenixedu.bennu.core.signals.Signal;
+import org.joda.time.LocalDate;
+
+@WebListener
+public class FenixEduISTSummariesContextListener implements ServletContextListener {
+
+    @Override
+    public void contextInitialized(ServletContextEvent sce) {
+        Signal.register(Summary.CREATE_SIGNAL, (DomainObjectEvent<Summary> event) -> {
+            Calendar gradeSubmissionEndDate = event.getInstance().getExecutionCourse().getExecutionDegrees().stream()
+                    .flatMap(ed -> ed.getPeriods(OccupationPeriodType.GRADE_SUBMISSION)).map(o -> o.getEndDate())
```

Comment:
What happens if there is no period defined for a certain degree.
This may produce a NullPointerException.